pkg/certificateprovider/inline/provider.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inline
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+)
+
+const (
+	// ValueParameter is the name of the parameter that contains the certificate (chain) as a string in PEM format
+	ValueParameter = "value"
+)
+
+func GetCertificates(ctx context.Context, attrib map[string]string) ([]*x509.Certificate, error) {
+	var certs []*x509.Certificate
+
+	value, ok := attrib[ValueParameter]
+	if !ok {
+		return nil, fmt.Errorf("value parameter is not set")
+	}
+
+	block, rest := pem.Decode([]byte(value))
+	if block == nil && len(rest) > 0 {
+		return nil, fmt.Errorf("failed to decode pem block")
+	}
+
+	for block != nil {
+		if block.Type == "CERTIFICATE" {
+			cert, err := x509.ParseCertificate(block.Bytes)
```

Comment:
I combined that logic and `bytesToCerts` (and tests) and moved it into the `certificateprovider` package. So now providers take care of all things related to certs, which makes sense. And the manager gets to stay focused on its role of handling k8s CRD events

This also made the function signatures line up consistently, rather than having the manager have to know about details of KeyVault types for further processing

Seems like it's in a good place to implement a factory, per the comment below